signatures.md

```diff
@@ -52,163 +49,350 @@ Note: `Message` is used here as the object being signed, but this interface shou
 }
 ```
 
+### Signed Message
 
-# Selected Signature Scheme
+This is the container for a signature that combines a message with an address and signature. 
+
+```go
+Type SignedMessage Struct {
 
-Currently, Filecoin uses secp256k1 signatures to fulfill the above interface. All signatures on messages, blocks, and tickets currently use the same scheme and format.
+		// Addrs is an array of Addresses associated with a particular message in Msgs 
+  	// at the same index. Each Address tracks the ID of the actor responsible for 
+  	// signing a particular Msg. An ID can be converted into a public key via 
+  	// Address.GetPublicKey(ID).
+		//
+		Addrs []Address `JSON`
+
+  	// Msgs is an array of messages. Each message in Msgs is represented by raw bytes.
+  	// Each message in Msgs will correspond to an Address in Addrs by index. 
+  	//
+    Msgs []bytes `JSON`   
+
+  	// Sig is a cryptographic signature that was generated according to the type 
+  	// reflected in Address.
+    Sig SignatureBytes `JSON` 
+    
+}
+```
 
-## Wire Format
+## Key Generation & Elliptic Curve Parameters
 
-What bits are on the wire and in what order/format. We are currently adhering to libsecp256k1 serialization which is laid out as follows. Note that this format description may not be accurate. See the github link below for an authoritative format description.
+This specification assumes the implementer uses a strong cryptographic pseudorandom generator `RANDOM`. The following was adapted from [IETF BLS Signature Scheme - Keygen](https://tools.ietf.org/html/draft-boneh-bls-signature-00#section-2.2). 
 
-**Signature**
+```go
+Input: RANDOM
+Output: PK, SK
 
+   1.  SK = x, chosen as a random integer in the range 1 and r-1
+   2.  PK = x*P, P is an element of an elliptic curve E
+   3.  Output PK, SK
 ```
-sig SignatureBytes = [0x30][len][0x02][r][indicator][s][indicator][recovery]
+
+For ECDSA, `P` is the generator for `E`. For BLS, we choose `P` to be a generator in `E2` which is the larger of the elliptic curve groups.
+
+### BLS12-381 Parameter Choices
+
+We have chosen to use the same elliptic curve parameters as [zkcrypto/pairing](https://github.com/zkcrypto/pairing/tree/master/src/bls12_381) with the exception that we are signing in G1 and generating public keys in G2. Note: there is a namespace collision between [zkcrypto/pairing](https://github.com/zkcrypto/pairing/tree/master/src/bls12_381) and [IETF BLS Signature Scheme](https://tools.ietf.org/html/draft-boneh-bls-signature-00) concerning G1 and G2. In zkcrypto/paring, G1 and G2 are points defined in the largest prime order subgroup of the elliptic curve defined by BLS12-381. In the IETF BLS Signature Scheme, G1 and G2 represent the largest prime order subgroups in BLS12-381. We will comply with the IETF standard. We will denote the generators of G1 and G2 as P1 and P2 respectively. E1 and E2 will denote the elliptic curve defined by BLS12-381.
```

Comment:
> G1 and G2 are points defined in the largest prime order subgroup

This is not quite as I read this: https://github.com/zkcrypto/pairing/tree/master/src/bls12_381#generators
To quote
> The generators of G1 and G2 are computed by finding the lexicographically...

So the use `G1` and `G2` for both the subgroup, as well as the generators.

Reply:
@dignifiedquire: You are correct. This is a language difference between the IETF standard and ZK-crypto. We are using the language from the IETF standard and not ZK-crypto. I'm going to leave the language as is.